lib/main.rb

```diff
@@ -0,0 +1,24 @@
+ALPHABET = ('a'..'z').to_a.freeze
```

Comment:
Per the assignment: Don’t forget to keep the same case.
Does this logic cover letters in upper case and lower case?

Reply:
Yeah! 

'What a string!'  --> 'Wmfy f xywnsl!'

Reply:
The W should be inverted too

Reply:
:O that's right jajaja

Reply:
Done!

'Bmfy f xywnsl!'

Reply:
```
  string.split('').map do |native_char|
    char = native_char.downcase ##### we'll use downcase...
    result = memory[char]
    result ||= ALPHABET.include?(char) ? change_letter(char, shift) : char
    memory[char] = result

    native_char != char ? result.upcase! : result  ##### ... and upcase if native_char is upercase
  end.join
```